so-widgets-bundle.php

```diff
@@ -98,37 +102,56 @@ function plugin_version_check(){
 
 		$active_version = get_option( 'siteorigin_widget_bundle_version' );
 
-		if( empty($active_version) || version_compare( $active_version, SOW_BUNDLE_VERSION, '<' ) ) {
-			// If this is a new version, then clear the cache.
-			update_option( 'siteorigin_widget_bundle_version', SOW_BUNDLE_VERSION );
+		$is_new = empty($active_version) || version_compare( $active_version, SOW_BUNDLE_VERSION, '<' );
+		$is_new = apply_filters( 'siteorigin_widgets_is_new_version', $is_new );
 
-			// Remove all cached CSS for SiteOrigin Widgets
-			if( function_exists('WP_Filesystem') && WP_Filesystem() ) {
-				global $wp_filesystem;
-				$upload_dir = wp_upload_dir();
-
-				// Remove any old widget cache files, if they exist.
-				$list = $wp_filesystem->dirlist( $upload_dir['basedir'] . '/siteorigin-widgets/' );
-				if( !empty($list) ) {
-					foreach($list as $file) {
-						// Delete the file
-						$wp_filesystem->delete( $upload_dir['basedir'] . '/siteorigin-widgets/' . $file['name'] );
-					}
-				}
-			}
+		if( $is_new ) {
 
-			// An action to let widgets handle the updates.
+			update_option( 'siteorigin_widget_bundle_version', SOW_BUNDLE_VERSION );
+			// If this is a new version, then trigger an action to let widgets handle the updates.
 			do_action( 'siteorigin_widgets_version_update', SOW_BUNDLE_VERSION, $active_version );
```

Comment:
A `clear_widget_cache` here wouldn't hurt. Just as an extra super backup for the other actions not firing for some reason. Or we could just hook another clear widget cache to the `siteorigin_widgets_version_update` action.